src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs

```diff
@@ -651,26 +651,38 @@ public void Cancel()
     {
         _threadingContext.ThrowIfNotOnUIThread();
 
-        // This wait is safe.  We are not passing the async callback to DismissUIAndRollbackEditsAndEndRenameSessionAsync.
-        // So everything in that method will happen synchronously.
-        DismissUIAndRollbackEditsAndEndRenameSessionAsync(
-            RenameLogMessage.UserActionOutcome.Canceled, previewChanges: false).Wait();
+        DismissUIAndRollbackEditsAndEndRenameSession(
+            RenameLogMessage.UserActionOutcome.Canceled, previewChanges: false);
     }
 
-    private async Task DismissUIAndRollbackEditsAndEndRenameSessionAsync(
+    private void DismissUIAndRollbackEditsAndEndRenameSession(
+        RenameLogMessage.UserActionOutcome outcome,
+        bool previewChanges)
+    {
+        _threadingContext.ThrowIfNotOnUIThread();
+
+        // Because we are passing null for finalSolution, we know this must complete synchronously.
+        DismissUIAndRollbackEditsAndEndRenameSession_MustBeCalledOnUIThreadAsync(outcome, previewChanges, finalSolution: null).VerifyCompleted();
+    }
+
+    /// <summary>
+    /// Dismisses the UI, rolls back any edits, and ends the rename session. If a <paramref name="finalSolution"/> is
+    /// provided, the workspace will be updated to its contents.  If <see langword="null"/> is pass for <paramref
+    /// name="finalSolution"/>, no change will be made to the workspace, and the call will complete synchronously after
+    /// resetting the UI state to what it was when the session started.
+    /// </summary>
+    private async Task DismissUIAndRollbackEditsAndEndRenameSession_MustBeCalledOnUIThreadAsync(
```

Comment:
💡 Async methods are generally expected to not make assertions about the calling thread. The calling pattern would be better adhered to by replacing `ThrowIfNotOnUIThread` with a call to `SwitchToMainThreadAsync`, which completes synchronously when the caller is already on the main thread. There would be no negative impact of switching to this pattern.